packages/core/src/payment/strategies/googlepay/googlepay-braintree-initializer.ts

```diff
@@ -59,24 +59,27 @@ export default class GooglePayBraintreeInitializer implements GooglePayInitializ
     }
 
     private _createGooglePayPayload(
-        checkout: Checkout,
+        checkout: Checkout | undefined,
         initializationData: any,
         hasShippingAddress: boolean
     ): GooglePayPaymentDataRequestV2 {
         if (!initializationData.platformToken) {
             throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
         }
 
+        const currencyCode = checkout?.cart.currency.code || '';
+        const totalPrice = checkout?.outstandingBalance ? round(checkout.outstandingBalance, 2).toFixed(2) : '';
+
         const googlePayBraintreePaymentDataRequest: GooglePayBraintreeDataRequest = {
             merchantInfo: {
                 authJwt: initializationData.platformToken,
                 merchantName: initializationData.googleMerchantName,
                 merchantId: initializationData.googleMerchantId,
             },
             transactionInfo: {
-                currencyCode: checkout.cart.currency.code,
+                currencyCode,
```

Comment:
Is it ok to pass empty string as currencyCode here? Is this data valuable for google pay?

Reply:
We should somehow replace this variables instead of adding empty string as default value..

Reply:
We update these `currencyCode` and `totalPrice` variables by `updatePaymentDataRequest` in `googlepay-payment-processor.ts`
You can see how it work here [googlepay-button-strategy.ts](https://github.com/bigcommerce/checkout-sdk-js/pull/1694/files/d13800a2cb62eead6f595da1cd8fe29669afc0c5#diff-6038e26677150a6bf876dab81fdd6b549ec356b2fa40212ff2c2b18cf6c18138R176)